pages/api/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { Prisma, PrismaClient } from "@prisma/client";
+import bcrypt from 'bcryptjs';
+import { NextApiRequest, NextApiResponse } from "next";
+
+const prisma = new PrismaClient();
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse){
+    if (req.method === 'POST'){
+        const {username, password} = req.body
+
+        try {
+
+            const user = await prisma.user.findUnique({where: {username}});
+            if (!user){
+                console.error('Invalid Username or Password');
+                res.status(401).json({ message: 'Incorrect Password' });
+            }
+            const passwordValid = await bcrypt.compare(password, user?.password ?? "");
+            if (!passwordValid){
+                console.error('Incorrect Password');
+                res.status(401).json({ message: 'Incorrect Password' });
+            }
+            res.status(200).json({ message: 'Sign in successful', user });
```

Comment:
This format looks reasonable, but I would still recommend conforming to an estabilished JSON response spec. See https://stackoverflow.com/questions/12806386/is-there-any-standard-for-json-api-response-format. I like [JSend](https://github.com/omniti-labs/jsend) myself because it's simple.